src/main/java/com/example/demo/Repository/EmailRepository.java

```diff
@@ -0,0 +1,52 @@
+package com.example.demo.Repository;
+
+
+import com.example.demo.Model.Email;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.MismatchedInputException;
+import org.springframework.stereotype.Repository;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Repository
+public class EmailRepository {
+
+    public boolean saveEmailsToFile(List<Email> list) {
+
+        Boolean result;
+        ObjectMapper mapper = new ObjectMapper();
+
+        try {
+            mapper.writeValue(new File("emails.json"), list );
+            result = true;
+        } catch (IOException e) {
+            e.printStackTrace();
+            result = false;
+        }
+
+        return result;
+    }
+    public List<Email> readEmailsFromFile()  {
```

Comment:
you can read/write file only once: at boot and shutdown time. During runtime it's more convenient to work with collection of objects. Then when application shutdowns you can register lifecycle callback to store collection into file